firebase-firestore/src/main/java/com/google/firebase/firestore/core/Transaction.java

```diff
@@ -53,6 +53,12 @@
   private final ArrayList<Mutation> mutations = new ArrayList<>();
   private boolean committed;
 
+  /**
+   * An error that may have occurred as a consequence of a write. If set, the task will fail instead
```

Comment:
"write" is actually a thing we can do outside of committing (and isn't a method actually in this class) so this comment is somewhat confusing. Additionally the point isn't to capture arbitrary errors so much as usage errors that that can't exposed directly at the call site.

Maybe something like this?

> A deferred usage error that occurred previously in this transaction that will cause the transaction to fail once it actually commits.